packages/shared/src/store/activityActions.ts

```diff
@@ -6,93 +6,52 @@ import * as ub from '../urbit';
-    db.setChannelVolumes([
-      { channelId: channel.id, isMuted: true, isNoisy: false },
-    ]);
-  }
+  const initialSettings =
+    channel.volumeSettings ?? (await db.getVolumeSettings(channel.id));
+
+  db.setVolumes([
+    {
+      itemId: channel.groupId ?? channel.id,
+      itemType: channel.groupId ? 'group' : 'channel',
+      isMuted: true,
+      isNoisy: false,
+    },
+  ]);
 
   try {
+    const { source } = api.getRootSourceFromChannel(channel);
+    const volume = ub.getVolumeMap('soft', true);
     await api.adjustVolumeSetting(source, volume);
   } catch (e) {
     logger.log(`failed to mute group ${channel.id}`, e);
     // revert the optimistic update
-    db.mergeVolumeSettings([
-      { sourceId, volume: existingSettings[sourceId] ?? null },
-    ]);
-    if (channel.groupId) {
-      db.setGroupVolumes([
-        {
-          groupId: channel.groupId,
-          isMuted: channel.isMuted ?? undefined,
-          isNoisy: channel.isNoisy ?? undefined,
-        },
-      ]);
-    } else {
-      db.setChannelVolumes([
-        {
-          channelId: channel.id,
-          isMuted: channel.isMuted ?? undefined,
-          isNoisy: channel.isNoisy ?? undefined,
-        },
-      ]);
+    if (initialSettings) {
+      await db.setVolumes([initialSettings]);
     }
   }
 }
 
 export async function unmuteChat(channel: db.Channel) {
-  const { source, sourceId } = api.getRootSourceFromChannel(channel);
-  const existingSettings = await db.getVolumeSettings();
-
-  // optimistic update
-  db.mergeVolumeSettings([{ sourceId, volume: null }]);
-
-  if (channel.groupId) {
-    db.setGroupVolumes([
-      { groupId: channel.groupId, isMuted: false, isNoisy: false },
-    ]);
-  } else {
-    db.setChannelVolumes([
-      { channelId: channel.id, isMuted: false, isNoisy: false },
-    ]);
-  }
+  const initialSettings =
+    channel.volumeSettings ?? (await db.getVolumeSettings(channel.id));
+
+  db.setVolumes([
+    {
+      itemId: channel.groupId ?? channel.id,
+      itemType: channel.groupId ? 'group' : 'channel',
+      isMuted: false,
+      isNoisy: false,
+    },
+  ]);
 
   try {
+    const { source } = api.getRootSourceFromChannel(channel);
     await api.adjustVolumeSetting(source, null);
   } catch (e) {
     logger.log(`failed to unmute chat ${channel.id}`, e);
     // revert the optimistic update
-    db.mergeVolumeSettings([
-      { sourceId, volume: existingSettings[sourceId] ?? null },
-    ]);
-
-    if (channel.groupId) {
-      db.setGroupVolumes([
-        {
-          groupId: channel.groupId,
-          isMuted: channel.isMuted ?? undefined,
-          isNoisy: channel.isNoisy ?? undefined,
-        },
-      ]);
-    } else {
-      db.setChannelVolumes([
-        {
-          channelId: channel.id,
-          isMuted: channel.isMuted ?? undefined,
-          isNoisy: channel.isNoisy ?? undefined,
-        },
-      ]);
+    if (initialSettings) {
+      await db.setVolumes([initialSettings]);
```

Comment:
We should talk quick. I _think_ we need to handle group vs channel conditionals when rolling back to existing values